File: src/java_answer/Lessons12921.java
package java_answer;

import static java.lang.System.*;

/**
 * @author dave.th
 * 소수찾기 : https://programmers.co.kr/learn/courses/30/lessons/12921
 */
public class Lessons12921 {

    public static void main(String[] args) {
        out.println((new Lessons12921()).solution(10));
    }

    private int solution(int n) {
        int answer = 0;
        for(int i=2; i<=n; i++) {
            boolean r = true;
            for(int j=2; j <= Math.sqrt(i); j++){
                if(i%j == 0) {
                    r = false;
                    break;
                }
            }
            if(r) answer++;
        }
        return answer;
    }
}

File: src/scala_answer/Lessons12921.scala
package scala_answer
import scala.util.control.Breaks._


/**
  * @author dave.th
  * 소수찾기 : https://programmers.co.kr/learn/courses/30/lessons/12921
  */
object Lessons12921 {
  def main(args: Array[String]) : Unit = {
    println(solution(10))
  }

  def solution(n : Int) : Int = {
    var answer = 0
    for(i <- 2 to n) {
      var r = true
      val sartI = Math.sqrt(i).toInt
      breakable {
        for(j <- 2 to sartI) {
          if(i%j == 0) {
            r = false
            break
          }
        }
      }
      if(r) answer += 1
    }

    answer
  }
}
